common/views/components/WorkTags/WorkTags.js

```diff
@@ -0,0 +1,85 @@
+import styled from 'styled-components';
+import { spacing, font, classNames } from '../../../utils/classnames';
+import NextLink from 'next/link';
+import type { NextLinkType } from '@weco/common/model/next-link-type';
+
+export type WorkTagType = {|
+  query: string,
+  textParts: string[],
+  linkAttributes: NextLinkType,
+|};
+
+const WorkTag = styled.div`
+  border-radius: 3px;
+  text-decoration: none;
+  padding: 0.2em 0.5em;
+  transition: color 250ms ease, background 250ms ease;
+`;
+
+type Props = {
+  tags: WorkTagType[],
+};
+
+const WorkTags = ({ tags }: Props) => {
+  return (
+    <ul
+      className={classNames({
+        'plain-list': true,
+        [spacing({ s: 1 }, { margin: ['top'] })]: true,
+        [spacing({ s: 0 }, { padding: ['left'] })]: true,
+      })}
+    >
+      {tags.map(({ query, textParts, linkAttributes }) => {
+        return (
+          <li
+            key={query}
+            className={classNames({
+              'inline-block': true,
+            })}
+          >
+            <NextLink {...linkAttributes}>
+              <a>
+                <WorkTag
+                  className={classNames({
+                    [spacing({ s: 1 }, { margin: ['right'] })]: true,
+                    [spacing({ s: 2 }, { margin: ['bottom'] })]: true,
+                    'line-height-1': true,
+                    'inline-block bg-hover-green font-hover-white': true,
+                    'border-color-green border-width-1': true,
```

Comment:
Are we deciding that this is the right way, and if so, what's the logic of the grouping?
(chanting again)...

Reply:
My logic was less well thought through here – mostly just wanted to keep the line length down a bit(!)